compiler/rustc_passes/src/diagnostic_items.rs

```diff
@@ -91,9 +91,21 @@ fn collect_item(
 
 /// Extract the first `rustc_diagnostic_item = "$name"` out of a list of attributes.
 fn extract(sess: &Session, attrs: &[ast::Attribute]) -> Option<Symbol> {
-    attrs.iter().find_map(|attr| {
-        if sess.check_name(attr, sym::rustc_diagnostic_item) { attr.value_str() } else { None }
-    })
+    let mut result = None;
+    for attr in attrs {
+        if sess.check_name(attr, sym::rustc_diagnostic_item) {
+            match attr.value_str() {
+                None => span_bug!(attr.span, "expected string value"),
```

Comment:
Shouldn't this be a diagnostic instead of a bug?
```suggestion
                None => sess.span_err(attr.span, "expected string value"),
```

Reply:
it's a compiler internal attribute, but since we do have a session available, using the error system doesn't hurt

Reply:
The difference is that `span_bug!` crashes the compiler immediately, while `span_err` goes on to find other errors.

Reply:
Sure. I don't how to add such a test case though.

Reply:
Actually I might be able to figure it out...